modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/application/ApplicationResource.java

```diff
@@ -128,6 +129,29 @@ public ListApplicationJson list( @QueryParam("query") final String query )
     }
 
     @GET
+    @Path("listKeys")
+    public List<ApplicationKeyJson> listKeys( @QueryParam("query") final String query )
+    {
+        Applications applications = this.applicationService.getInstalledApplications();
+        if ( StringUtils.isNotBlank( query ) )
+        {
```

Comment:
This block can be extracted in a separate filterApplications method, is already used in 3 places in this class.